packages/framework/src/Model/Product/ProductPromotionXyData.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Product;
+
+class ProductPromotionXyData
+{
+    /**
+     * @var int|null
+     */
+    public $buyQuantity;
+
+    /**
+     * @var int|null
+     */
+    public $freeQuantity;
+
+    public function __construct()
+    {
+        $this->buyQuantity = null;
```

Comment:
null is inherently set as a default value if the property does not has an explicit type